contracts/adapters/OpenseaDownpaymentBuyAdapter.sol

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.4;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IAaveFlashLoanReceiver} from "./interfaces/IAaveFlashLoanReceiver.sol";
+import {ILendPool} from "../interfaces/ILendPool.sol";
+import {IOpenseaExchage} from "./interfaces/IOpenseaExchage.sol";
+import {ILendPoolAddressesProvider} from "../interfaces/ILendPoolAddressesProvider.sol";
+import {IAaveLendPoolAddressesProvider} from "./interfaces/IAaveLendPoolAddressesProvider.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+import {IERC721Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+
+contract OpenseaDownpaymentBuyAdapter is
+  IAaveFlashLoanReceiver,
+  OwnableUpgradeable,
+  PausableUpgradeable,
+  ReentrancyGuardUpgradeable
```

Comment:
need inherit ERC721Holder, for safeTransferFrom to addres(this)

Reply:
done